src/test/java/com/commercetools/sync/customers/helpers/CustomerReferenceResolverTest.java

```diff
@@ -0,0 +1,352 @@
+            .of("email@example.com", "secret123")
+            .customerGroup(ResourceIdentifier.ofKey("anyKey"))
+            .key("dummyKey");
+
+        final CompletableFuture<Optional<String>> futureThrowingSphereException = new CompletableFuture<>();
+        futureThrowingSphereException.completeExceptionally(new SphereException("CTP error on fetch"));
+        when(customerGroupService.fetchCachedCustomerGroupId(anyString())).thenReturn(futureThrowingSphereException);
+
+        assertThat(referenceResolver.resolveCustomerGroupReference(customerDraftBuilder))
+            .hasFailedWithThrowableThat()
+            .isExactlyInstanceOf(SphereException.class)
+            .hasMessageContaining("CTP error on fetch");
+    }
+
+    @Test
+    void resolveCustomerGroupReference_WithIdOnCustomerGroupReference_ShouldNotResolveReference() {
+        final CustomerDraftBuilder customerDraftBuilder = CustomerDraftBuilder
+            .of("email@example.com", "secret123")
+            .customerGroup(ResourceIdentifier.ofId("existingId"))
+            .key("dummyKey");
+
+        assertThat(referenceResolver.resolveCustomerGroupReference(customerDraftBuilder).toCompletableFuture())
+            .hasNotFailed()
+            .isCompletedWithValueMatching(resolvedDraft -> Objects.equals(resolvedDraft.getCustomerGroup(),
+                customerDraftBuilder.getCustomerGroup()));
+    }
+
+    @Test
+    void resolveStoreReferences_WithNullStoreReferences_ShouldNotResolveReferences() {
+        final CustomerDraftBuilder customerDraftBuilder = CustomerDraftBuilder
+            .of("email@example.com", "secret123")
+            .stores(null)
+            .key("dummyKey");
+
+        final CustomerDraftBuilder resolvedDraft = referenceResolver
+            .resolveStoreReferences(customerDraftBuilder)
+            .toCompletableFuture()
+            .join();
+
+        assertThat(resolvedDraft.getStores()).isNull();
+    }
+
+    @Test
+    void resolveStoreReferences_WithEmptyStoreReferences_ShouldNotResolveReferences() {
+        final CustomerDraftBuilder customerDraftBuilder = CustomerDraftBuilder
+            .of("email@example.com", "secret123")
+            .stores(emptyList())
+            .key("dummyKey");
+
+        final CustomerDraftBuilder resolvedDraft = referenceResolver
+            .resolveStoreReferences(customerDraftBuilder)
+            .toCompletableFuture()
+            .join();
+
+        assertThat(resolvedDraft.getStores()).isEmpty();
+    }
+
+    @Test
+    void resolveStoreReferences_WithValidStores_ShouldResolveReferences() {
+        final CustomerDraftBuilder customerDraftBuilder = CustomerDraftBuilder
+            .of("email@example.com", "secret123")
+            .stores(asList(ResourceIdentifier.ofKey("store-key1"),
+                ResourceIdentifier.ofKey("store-key2"),
+                ResourceIdentifier.ofId("store-id-3")))
+            .key("dummyKey");
+
+        // test
```

Comment:
I see sometimes you have `//test` and `// assertion`, sometimes you don't. I would have 1 style - either have it everywhere or nowhere. In this file, the tests are pretty easy, so it's not even necessary.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/587/commits/ebf3d30834d883ab6fb714a8bcc2756d65b19ff3